test/jdk/java/net/httpclient/ServerCloseTest.java

```diff
@@ -287,77 +287,83 @@ public void run() {
+                            clientConnection.getOutputStream(), UTF_8);
                     PrintWriter pw = new PrintWriter(w);
                     System.out.println(now() + getName() + ": Reading request line");
                     String requestLine = readLine(ccis);
                     System.out.println(now() + getName() + ": Request line: " + requestLine);
 
                     StringTokenizer tokenizer = new StringTokenizer(requestLine);
-                    String method = tokenizer.nextToken();
-                    assert method.equalsIgnoreCase("POST")
-                            || method.equalsIgnoreCase("GET");
+                    tokenizer.nextToken(); // Skip method token as not used
                     String path = tokenizer.nextToken();
-                    URI uri;
+
+                    URI uri = null;
+                    boolean validURI = true;
                     try {
                         String hostport = serverAuthority();
-                        uri = new URI((secure ? "https" : "http") +"://" + hostport + path);
+                        uri = new URI((secure ? "https" : "http") + "://" + hostport + path);
                     } catch (Throwable x) {
                         System.err.printf("Bad target address: \"%s\" in \"%s\"%n",
                                 path, requestLine);
-                        clientConnection.close();
-                        continue;
-                    }
-
-                    // Read all headers until we find the empty line that
-                    // signals the end of all headers.
-                    String line = requestLine;
-                    while (!line.equals("")) {
-                        System.out.println(now() + getName() + ": Reading header: "
-                                + (line = readLine(ccis)));
-                        headers.append(line).append("\r\n");
+                        validURI = false;
                     }
 
-                    StringBuilder response = new StringBuilder();
-
-                    int index = headers.toString()
-                            .toLowerCase(Locale.US)
-                            .indexOf("content-length: ");
-                    byte[] b = uri.toString().getBytes(UTF_8);
-                    if (index >= 0) {
-                        index = index + "content-length: ".length();
-                        String cl = headers.toString().substring(index);
-                        StringTokenizer tk = new StringTokenizer(cl);
-                        int len = Integer.parseInt(tk.nextToken());
-                        assert len < b.length * 2;
-                        System.out.println(now() + getName()
-                                + ": received body: "
-                                + new String(ccis.readNBytes(len), UTF_8));
+                    // Proceed if URI is well-formed and the request path is as expected
+                    if (validURI && path.contains("/dummy/x")) {
```

Comment:
Since we will have already checked `path`, `method` and `uri`  here just unconditionally add the clientConnection to the list and proceed as before.